### PR TITLE
add --version option to the rootCmd

### PR DESCRIPTION
**Pull request by N4r35h**
resolves issue #298 

its already built into cobra just needed to add the Version string to the rootCommand

image

### CODE REVIEW

**Comment by guergabo** [comment]
Hi @N4r35h, 

Thanks for this! I was thinking of using go's build process for this issue instead and pass the versioning information to the linker via `ldflags`. That way it's not so static and we can configure it externally from a github workflow. Otherwise, we have to modify the source code for each build.

It could default to dev or something of the sorts. 

Feel free to explore this route and update this PR. Thanks.

**Comment by N4r35h** [comment]
sure @guergabo,
i have made the changes please refer to the image and let me know if we should add something else or change things around

i added the vsc.version (automatically set by go build) which is the commit hash also feeling it might be useful where we distribute dev versions of the build for testing and want to correlate to the particular commit also seen it done by the docker cli, we can add more info such as version of golang it was compiled with and built time if required

could you explain about the github release workflow or confirm if thats being used right now ?

`go build -ldflags="-X github.com/resonatehq/resonate/cmd/version.Version=v0.5.0"`

image

File: cmd/version/version.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"fmt"
+	"runtime/debug"
+
+	"github.com/spf13/cobra"
+)
+
+var Version string = "dev"
+
+var VersionCmd = &cobra.Command{
+	Use: "version",
```
Comment (guergabo) [inline cmd/version/version.go]: no need to make a whole command out of this a variable should do. check this out: https://www.digitalocean.com/community/tutorials/using-ldflags-to-set-version-information-for-go-applications

File: cmd/root.go [hunk 2]
```diff
@@ -42,6 +43,7 @@ func init() {
 	rootCmd.AddCommand(dst.NewCmd())
 	rootCmd.AddCommand(serve.ServeCmd())
 	rootCmd.AddCommand(quickstart.NewCmd())
+	rootCmd.AddCommand(version.VersionCmd)
```
Comment (guergabo) [inline cmd/root.go]: the behavior should still remain using the cobra version flag: `resonate --version`

**Review by guergabo — CHANGES_REQUESTED** [review]

**Comment by guergabo** [comment]
regarding the github release workflow, it starts here: https://github.com/resonatehq/resonate/blob/main/.github/workflows/release.yaml

let me take a look in the morning and suggest some follow up actions, because we would want the version information passed to both our `docker image release build` & `github release artifact build`

**Comment by guergabo** [comment]
ok, for docker image release build:
- most likely `--build-arg` should do the job and use environment variables (the ARG keyword) for this build: https://github.com/resonatehq/resonate/blob/main/Dockerfile#L7

for github release build: 
- we just switched to nix and the ldflags are passed here: https://github.com/resonatehq/resonate/blob/main/flake.nix#L88 
- for this i want to defer to @lucperkins to suggest the best approach to pass an argument.

**Comment by N4r35h** [comment]
thanks for pointing out the flake.nix file

i have made the requested changes consider it a draft let me know if anything else is required
i will look into setting this into the workflow in a while

File: cmd/root.go [hunk 3]
```diff
@@ -17,11 +19,27 @@ import (
 
 var (
 	server, cfgFile string
+	Version         string = "dev"
 )
 
+func computeVersion() string {
```
Comment (guergabo) [inline cmd/root.go:25]: check out the link i sent before, i'll paste here again: https://www.digitalocean.com/community/tutorials/using-ldflags-to-set-version-information-for-go-applications

this whole function is not needed.

**Review by guergabo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**